pkg/repository/ghreleases/ghreleases.go

```diff
@@ -82,9 +82,54 @@ func (ghr *GHReleases) PublishProbe(driverVersion string, probePath string) erro
 }
 
 // IsAlreadyMirrored implmements repository.Repository.IsAlreadyMirrored for GitHub Releases.
-func (ghr *GHReleases) IsAlreadyMirrored(driverVersion string, probeName string) bool {
-	// TODO: unimplimented
-	return false
+func (ghr *GHReleases) IsAlreadyMirrored(driverVersion string, probeName string) (bool, error) {
+	// use a mutex to ensure that this function is only called once at a time as it is not thread safe.
+	// i.e. a release that doesn't exist may result in multiple goroutines trying to create it at once.
```

Comment:
I don't think you need a mutex here (and the "i.e" doesn't fit with what this function is doing) as you're only listing releases.

Reply:
I've removed the mutex and the comment.